core/src/main/java/org/apache/accumulo/core/clientImpl/AccumuloClientImpl.java

```diff
@@ -298,82 +311,93 @@ private ClientInfo getClientInfo() {
     }
 
     @Override
-    public AccumuloClient build() throws AccumuloException, AccumuloSecurityException {
+    public T build() {
+      return builderFunction.build(this);
+    }
+
+    public static ClientBuilderImpl<AccumuloClient> newClientBuilder() {
+      return new ClientBuilderImpl<>(ClientBuilderImpl::buildClient);
+    }
+
+    public static ClientBuilderImpl<Properties> newPropertiesBuilder() {
+      return new ClientBuilderImpl<>(ClientBuilderImpl::buildProps);
+    }
+
+    private static AccumuloClient buildClient(ClientBuilderImpl<AccumuloClient> cbi) {
       SingletonReservation reservation = SingletonManager.getClientReservation();
       try {
-        return new AccumuloClientImpl(reservation, new ClientContext(getClientInfo()));
-      } catch (AccumuloException | AccumuloSecurityException | RuntimeException e) {
+        return new AccumuloClientImpl(reservation, new ClientContext(cbi.getClientInfo()));
+      } catch (RuntimeException e) {
```

Comment:
This could use a comment explaining that AccumuloClientImpl is responsible for closing the reservation if it is constructed successfully. Otherwise, this code looks like it could be cleaned up using try-with-resources (which it cannot be, because we only want to close the reservation on RTE).

Reply:
Done